### DISCUSSION TITLE
different environment with different parameters

### DISCUSSION THREAD

**Discussion by yonglee7015** [pymupdf/PyMuPDF#2368] [Q&A]
clip_rect = Rect(size_word[0][0] - 10, size_word[0][1] - **25**
                                 , size_word[0][2] + 10, size_word[0][1])

you can notice that I am setting a Rect Based on a size_word location.
It works fine in my local machine. But after I deploy it to a web-server(pythonanywhere), I need to adjust 25 to 20 or 15. Why?
same code in different environment need different parameters?

**Comment by JorjMcKie** [comment]
Are you sure these are the files being worked on? What if a word is written 90° rotated? It that case, the height of the rectangle corresponds to the written length of the word ...

**Reply by yonglee7015** [reply]
Yes , sure. I test **same pdf files** with same code, just in different environment. So it is not relevant to rotation.

**Reply by JorjMcKie** [reply]
Hm, weird. Can you let me have a file example please? The problem - if any - must be rooted in the original extracted coordinates of the word.
And that in turn is hard to believe - at least if the PDF has the respective font embedded.
A non-embedded font (showing "n/a" in `page.get_fonts()`), **_might_** open a way to explain this, although I wouldn't know how at the moment.

**Reply by yonglee7015** [reply]
Hi JorjMcKie, you can test it with any pdf files , write a simple code, extract a text based on a text coordinates. then upload it to pythonanywhere.  Run it  in pythonanywhere, you will notice that you need to adjust the parameters.

```
# find sample size based on keyword, fitz module
def find_sample_size(pdf_folder, sizepage):
    """with fitz module,based on keywords, locate sample size positon on pdf
       search return a rect
       pdf_folder: string pdf document path
       sizepage: size table page
    """
    size_keyword = ['Base Size', 'Sample Size']
    sample_size = ''
    pdf = fitz.open(pdf_folder)
    page = pdf[sizepage]
    for i in range(len(size_keyword)):
        size_word = page.search_for(size_keyword[i])
        if size_word:
            if page.rotation == 90:
                clip_rect = Rect(size_word[0][0] - 10, size_word[0][1] - 25
                                 , size_word[0][2] + 10, size_word[0][1])
            else:
                clip_rect = Rect(size_word[0][2] + 5, size_word[0][1] - 10
                                 , size_word[0][2] + 150, size_word[0][3] + 10)
            sample_size = page.get_text('text', clip=clip_rect).strip(':').strip()
            break
    return sample_size
```

**Reply by JorjMcKie** [reply]
Sorry, I have no account there. Obviously, there are installation differences between there and your local machine, which you must explore - first and foremost the PyMuPDF versions.
